### PR TITLE
Avoid assembly load failure on Windows.

### PR DESCRIPTION
**Pull request by KirillOsenkov**
For some reason even though we call Assembly.Load from the codebase, it still doesn't seem to get promoted from LoadFrom to the Load context.

A solution is to use AssemblyResolve event and just resolve to the assembly that's already loaded.

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs [hunk 1]
```diff
@@ -486,6 +486,27 @@ static void LoadMSBuildLibraries ()
...
+			if (Type.GetType ("Mono.Runtime") == null) {
+				AppDomain.CurrentDomain.AssemblyResolve += CurrentDomain_AssemblyResolve;
+			}
+		}
+
+		/// <summary>
+		/// This is necessary on Windows because without it, even though the assemblies are already loaded
+		/// the CLR will still throw FileNotFoundException (unable to load assembly).
+		/// </summary>
+		private static System.Reflection.Assembly CurrentDomain_AssemblyResolve (object sender, ResolveEventArgs args)
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs]: Nit: redundant accessibility modifier.

Comment (KirillOsenkov) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs]: Thanks, fixed.

File: main/Main.sln [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26228.4
+VisualStudioVersion = 15.0.26403.0
```
Comment (Therzok) [inline main/Main.sln]: Woops :p

Comment (KirillOsenkov) [inline main/Main.sln]: Yeah reverted.

**Pull request merged**